dataprofiler/profilers/float_column_profile.py

```diff
@@ -106,16 +106,26 @@ def _get_float_precision(cls, df_series_clean):
         :return: string representing its precision print format
         :rtype: int
         """
-        if not len(df_series_clean):
-            return 0
+        if not len(df_series_clean): return 0
 
         # Lead zeros: ^[+-.0\s]+ End zeros: \.?0+(\s|$)
         # Scientific Notation: (?<=[e])(.*) Any non-digits: \D
         r = re.compile(r'^[+-.0\s]+|\.?0+(\s|$)|(?<=[e])(.*)|\D')
-        
-        return float(df_series_clean.replace(
-            to_replace=r, value='').map(len).min())
-        
+
+        # Take a sampling of the dataset. If small use full dataset,
+        # OR 20k samples or 5% of the dataset which ever is larger.
+        sample_size = min(len(df_series_clean),
```

Comment:
I'm not sure about sampling here. This is the only part of the entire repo where the user is forced to sample their dataset (we have options for profiler and data labeler sample size, so they could be set not to sample)